src/Raven.Server/Documents/Indexes/Persistence/Corax/AnonymousCoraxDocumentConverter.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Amazon.SimpleNotificationService.Model;
+using Corax;
+using NuGet.Protocol;
+using Raven.Client.Documents.Indexes;
+using Raven.Server.Documents.Indexes.Persistence.Corax.WriterScopes;
+using Raven.Server.Documents.Indexes.Persistence.Lucene.Documents;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server;
+using Raven.Server.Utils;
+using Constants = Raven.Client.Constants;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Corax;
+
+public class AnonymousCoraxDocumentConverter : CoraxDocumentConverterBase
+{
+    private readonly bool _isMultiMap;
+    private IPropertyAccessor _propertyAccessor;
+
+    public AnonymousCoraxDocumentConverter(Index index, bool storeValue = false)
+        : base(index, storeValue, false, true, 1, null,Constants.Documents.Indexing.Fields.ReduceKeyValueFieldName)
+    {
+        _isMultiMap = index.IsMultiMap;
+    }
+
+    public override void Dispose()
+    {
+        //throw new NotImplementedException();
+    }
+
+    public override Span<byte> SetDocumentFields(LazyStringValue key, LazyStringValue sourceDocumentId, object doc, JsonOperationContext indexContext,
+        out LazyStringValue id)
+    {
+        //Lucene leftover, we will not support boosting inside index.
+        var boostedValue = doc as BoostedValue;
+        var documentToProcess = boostedValue == null ? doc : boostedValue.Value;
+
+        IPropertyAccessor accessor;
+
+        if (_isMultiMap == false)
+            accessor = _propertyAccessor ??= PropertyAccessor.Create(documentToProcess.GetType(), documentToProcess);
+        else
+            accessor = TypeConverter.GetPropertyAccessor(documentToProcess);
+
+        using var _ = _allocator.Allocate(9086 * 8, out ByteString buffer);
+        var entryWriter = new IndexEntryWriter(buffer.ToSpan(), _knownFields);
+        
+        id = key ?? (sourceDocumentId ?? throw new InvalidParameterException(""));
+        List<int> stringsLength = new List<int>(128);
```

Comment:
Can we cache / reuse this?

Reply:
Yes, already changed. I reuse it for the whole indexing process.  

Notice that `EnumerableWriterScope` also needs cache, but there is a ticket to rebuild it:
https://issues.hibernatingrhinos.com/issue/RavenDB-17243
The current implementation is temporary and needs to be changed before the first release.